src/main/java/net/consensys/orion/storage/OrionSQLKeyValueStore.java

```diff
@@ -63,7 +65,8 @@ public Bytes get(final Bytes key, final Continuation<? super Bytes> ignore) {
   @Override
   public Unit put(final Bytes key, final Bytes value, final Continuation<? super Unit> ignore) {
     final Store store = new Store();
-    store.setKey(key.toArrayUnsafe());
+    final String b64String = Base64.getEncoder().encodeToString(key.toArrayUnsafe());
```

Comment:
Since the length is now limited to 120 characters, thinking we check that. Otherwise we will probably fail with some hard to understand database error.

Reply:
The error returned is pretty good:
Caused by: org.apache.openjpa.lib.jdbc.ReportingSQLException: ORA-12899: value too large for column "C##STEFAN"."STORE"."KEY" (actual: 60, maximum: 59)

Do you think we should still add the check? It would not do any harm, but the error message would not be any better than that. We might even want to change the key size from 120 to 60, as that is the size that we are using at the moment.

Reply:
Thinking about this more, it would be better to catch this even earlier before it got to the store so that a useful error could be returned on the API response. Is that possible?

Reply:
What if we just were to make it a varchar2, which has a maximum size of 4000? They are only storing the actual data, with some metadata that contains the actual size of a key. I read that it would be advantages to limit the size if you are using PL/SQL scripts if you know that your data will be limited to a small size, but that should not be of concern for us ...
Should we just specify varchar2(4000)? I don't like putting a check into our code, as the only limit that we have is in the DB ...